src/anemoi/graphs/plotting/prepare.py

```diff
@@ -0,0 +1,179 @@
+        Name of the target nodes.
+
+    Returns
+    -------
+    latitudes : np.ndarray
+        Latitude coordinates of the points connected.
+    longitudes : np.ndarray
+        Longitude coordinates of the points connected.
+    """
+    sub_graph = graph[(source_nodes_name, "to", target_nodes_name)].edge_index
+    x0 = np.rad2deg(graph[source_nodes_name].x[sub_graph[0]])
+    y0 = np.rad2deg(graph[target_nodes_name].x[sub_graph[1]])
+    nans = np.full_like(x0[:, :1], np.nan)
+    latitudes = np.concatenate([x0[:, :1], y0[:, :1], nans], axis=1).flatten()
+    longitudes = np.concatenate([x0[:, 1:2], y0[:, 1:2], nans], axis=1).flatten()
+    return latitudes, longitudes
+
+
+def compute_node_adjacencies(
+    graph: HeteroData, source_nodes_name: str, target_nodes_name: str, num_nodes: int
+) -> tuple[list[int], list[str]]:
+    """Compute the number of adjacencies of each node in a bipartite graph.
+
+    Parameters
+    ----------
+    graph : HeteroData
+        Graph to plot.
+    source_nodes_name : str
+        Name of the dimension of the coordinates for the head nodes.
+    target_nodes_name : str
+        Name of the dimension of the coordinates for the tail nodes.
+    num_nodes : int
+        Number of nodes in the destination (including the not connected).
+
+    Returns
+    -------
+    num_adjacencies : list[int]
+        Number of adjacencies of each node.
+    node_text : list[str]
+        Text to show when hovering over the nodes.
+    """
+    node_adjacencies = np.zeros(num_nodes, dtype=int)
+    vals, counts = np.unique(graph[(source_nodes_name, "to", target_nodes_name)].edge_index[1], return_counts=True)
+    node_adjacencies[vals] = counts
+    node_text = [f"# of connections: {x}" for x in node_adjacencies]
+    return list(node_adjacencies), node_text
+
+
+def compute_isolated_nodes(graph: HeteroData) -> dict[str, tuple[list, list]]:
+    """Compute the isolated nodes.
+
+    Parameters
+    ----------
+    graph : HeteroData
+        Graph.
+
+    Returns
+    -------
+    dict[str, list[int]]
+        Dictionary with the isolated nodes for each subgraph.
+    """
+    isolated_nodes = {}
+    for (source_name, _, target_name), sub_graph in graph.edge_items():
+        head_isolated = np.ones(graph[source_name].num_nodes, dtype=bool)
+        tail_isolated = np.ones(graph[target_name].num_nodes, dtype=bool)
+        head_isolated[sub_graph.edge_index[0]] = False
+        tail_isolated[sub_graph.edge_index[1]] = False
+        if np.any(head_isolated):
+            isolated_nodes[f"{source_name} isolated (--> {target_name})"] = node_list(
+                graph, source_name, mask=list(head_isolated)
+            )
+        if np.any(tail_isolated):
+            isolated_nodes[f"{target_name} isolated ({source_name} -->)"] = node_list(
+                graph, target_name, mask=list(tail_isolated)
+            )
+
+    return isolated_nodes
+
+
+def _get_node_attribute_dims(graph: HeteroData) -> dict[str, int]:
```

Comment:
Similarly between _get_node_attribute_dims and _get_edge_attribute_dims is there a better way to abstract this functionality to reduce the amount of repeated code?

Reply:
I also believe these shouldn't be private functions